### PR TITLE
#2979 - Email notification - PD/PPD Student reminder email 8 weeks before end date Part 3

### PR DESCRIPTION
**Pull request by bidyashish**
- [x] E2E Tests
  - [x] Ensure the email will be sent once for the same assessment.
  - [x] Ensure the email will be sent again for different assessments for the same application.

### CODE REVIEW

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/student-application-notifications/_tests_/student-application-notifications.scheduler.e2e-spec.ts [hunk 1]
```diff
@@ -110,5 +120,178 @@ describe(
...
+        // Queued job.
+        const mockedJob = mockBullJob<void>();
+
+        // Act
+        await processor.studentApplicationNotifications(mockedJob.job);
+
+        // Assert
+        expect(
+          mockedJob.containLogMessages([
+            `No assessments found to generate PD/PPD mismatch notifications.`,
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/student-application-notifications/_tests_/student-application-notifications.scheduler.e2e-spec.ts:180]: Minor: Please use double quotes.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/student-application-notifications/_tests_/student-application-notifications.scheduler.e2e-spec.ts [hunk 2]
```diff
@@ -110,5 +120,178 @@ describe(
...
+              metadata: {
+                assessmentId: application.currentAssessment.id,
+              },
+            },
+          },
+        );
+
+        await db.notification.save(studentNotification);
+
+        // Create a new assessment for the same application
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/student-application-notifications/_tests_/student-application-notifications.scheduler.e2e-spec.ts:245]: Minor: add a period at the end of the phrase. :)

**Review by andrepestana-aot — APPROVED** [review]
Looks good! Only minor comments.

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍

**Pull request merged**